controller/orphan_controller.go

```diff
@@ -227,13 +263,11 @@ func (oc *OrphanController) reconcile(orphanName string) (err error) {
 	}
 
 	if !orphan.DeletionTimestamp.IsZero() {
-		defer func() {
-			if err == nil {
-				err = oc.ds.RemoveFinalizerForOrphan(orphan)
-			}
-		}()
-
-		return oc.cleanupOrphanedData(orphan)
+		isCleanupComplete, err := oc.cleanupOrphanedResource(orphan)
```

Comment:
Is it possible `isCleanupComplete` is `false` but `err` is `nil`?

Reply:
Instance manager may take time to wait for instance terminating. Orphan controller will confirm the instance state with instance manager after sending the deletion request. `isCleanupComplete` will be false if the instance is still exist (terminating or some other state), and the deletion will continue after the instance state changes, triggered by instance manager status update event.